### PR TITLE
Introduce `start_page` argument to partitioning functions that assign `element.metadata.page_number`

### PR DESCRIPTION
**Pull request by micmarty-deepsense**
This small change will be useful for users who partition only fragments of their PDF documents. 
It's a small step towards addressing this issue: https://github.com/Unstructured-IO/unstructured/issues/2461

Related PRs:
* https://github.com/Unstructured-IO/unstructured/pull/2842
* https://github.com/Unstructured-IO/unstructured/pull/2673

### CODE REVIEW

**Review by scanny — COMMENTED** [review]
Looks good so far, I've added a couple small remarks.

On the `start_page` vs. `starting_page_number` question, after reflection, I think we should spell it out. As much as I like shorter names, and I do, I think being explicit is going to be the better choice here. Concise names are mostly about line-length, but this is a parameter, and when used in a call it's almost always going to be stacked anyway (because there are so many parameters) and it's only going to appear in a small number of expressions inside the code, so I think the longer form is the right choice.

Note that we'll need to add `starting_page_number` to DOCX, PPTX, and XLSX as well since they support page-numbers (although not fragmenting). Rationale being `.metadata.page_number` should reflect the specified starting page number, whether or not we fragment for performance. So all partitioners that populate page-number should be sensitive to this partitioning parameter.

File: unstructured/partition/common.py [hunk 1]
```diff
@@ -622,7 +623,7 @@ def document_to_element_list(
         if sortable and sort_mode != SORT_MODE_DONT:
             sorted_page_elements = sort_page_elements(page_elements, sort_mode)
 
-        if include_page_breaks and i < num_pages - 1:
+        if include_page_breaks and page_number < start_page + num_pages:
             sorted_page_elements.append(PageBreak(text=""))
```
[the remark this answers is not in the document]
Comment (micmarty-deepsense) [reply unstructured/partition/common.py:627]: 👍 let's do that in another PR, I already have a rough sketch locally

File: unstructured/partition/auto.py [hunk 2]
```diff
@@ -243,6 +244,10 @@ def partition(
         Applies only when providing file via `file` parameter. If this option is True and inference
         from message header failed, attempt to infer last_modified metadata from bytes,
         otherwise set it to None.
+    start_page
+        Indicates what page number should be assigned for the first page in the document.
```
[the remark this answers is not in the document]
Comment (micmarty-deepsense) [reply unstructured/partition/auto.py]: fixed😅

**Review by scanny — APPROVED** [review]
LGTM :)

**Pull request merged**